db/sqlinterfaces.c

```diff
@@ -4106,6 +4108,62 @@ __func__, cnonce);
 
 */
 
+int check_active_appsock_connections(struct sqlclntstate *clnt)
+{
+    int max_appsock_conns =
+        bdb_attr_get(thedb->bdb_attr, BDB_ATTR_MAXAPPSOCKSLIMIT);
+    if (active_appsock_conns > max_appsock_conns) {
+        int num_retry = 0;
+        int rc = -1;
+    retry:
+        Pthread_mutex_lock(&clnt_lk);
+        Pthread_mutex_lock(&appsock_conn_lk);
```

Comment:
Would it be possible to use one lock + `ATOMIC_ADD` for counter?

Reply:
active_appsock_conns needs to be under lock for this logic but got separate lock to reduce contention, as we will be inside this loop rarely.